fe/lib/components/Questionnaire/components/FormBuilderQueryInput.tsx

```diff
@@ -0,0 +1,70 @@
+import {
+  Actions,
+  Alert,
+  Button,
+  Card,
+  ContentBlock,
+  Stack,
+  Text,
+  Textarea,
+} from 'braid-design-system';
+import React, { useState } from 'react';
+
+import { MutationVariableInput, validateQueryInput } from '../questionTypes';
+
+interface FormBuilderQueryInputProps {
+  onChange: (variables: MutationVariableInput) => void;
+}
+
+export const FormBuilderQueryInput = ({
+  onChange,
+}: FormBuilderQueryInputProps) => {
+  const [input, setInput] = useState('');
+
+  const [queryInputError, setQueryInputError] = useState<string>('');
+  const [showError, setShowError] = useState(false);
+
+  const checkValidQuery = () => {
+    try {
+      const variables = validateQueryInput(JSON.parse(input));
+      onChange(variables);
+
+      setShowError(false);
+    } catch (err) {
+      setShowError(true);
+      if (err.name === 'ValidationError') {
+        setQueryInputError(`${err.name}: ${err.key} is invalid`);
+      } else {
+        setQueryInputError(`${err.name}: ${err.message}`);
+      }
+    }
+  };
+
+  return (
+    <ContentBlock>
+      <Stack space="gutter">
+        <Card>
+          <Stack space="gutter">
+            <Textarea
+              label="Graphql Variables"
+              id="id"
```

Comment:
This doesn't seem right. It should probably be parameterised and provided by the consuming app.

Reply:
Did you mean the value for the text area?

Reply:
This becomes a [global element ID in the DOM](https://developer.mozilla.org/en-US/docs/Web/API/Element/id) i.e. `<textarea id="id"></textarea>`, so it's not good for a reusable component to squat on a generic value like `id`.